ProcessLib/ThermoMechanics/ThermoMechanicsFEM.h

```diff
@@ -668,3 +741,5 @@ class ThermoMechanicsLocalAssembler
 
 }  // namespace ThermoMechanics
 }  // namespace ProcessLib
+
+#include "ThermoMechanicsFEM-impl.h"
```

Comment:
Either all implementations go into the `impl.h` file, or no. I prefer the latter.

Reply:
For an easy reviewing, I did not move the existing function definition  to impl.h. I planned to do it once this PR is merged.